File: frontend/src/api/document.ts
import request from '@/utils/request'

export interface DocumentItem {
  id: number
  filename: string
  title: string
  file_type: string
  file_size: number
  chunk_count: number
  status: string
  category: string
  graph_status: string
  graph_error: string | null
  created_at: string
}

export interface DocumentChunk {
  id: number
  document_id: number
  chunk_index: number
  content: string
  token_estimate: number
}

export interface DocumentDetail extends DocumentItem {
  chunks: DocumentChunk[]
}

export function uploadDocument(file: File, category?: string): Promise<DocumentItem> {
  const form = new FormData()
  form.append('file', file)
  if (category) form.append('category', category)
  return request.post('/documents', form, {
    headers: { 'Content-Type': 'multipart/form-data' },
  })
}

export function listDocuments(): Promise<DocumentItem[]> {
  return request.get('/documents')
}

export function getDocument(id: number): Promise<DocumentDetail> {
  return request.get(`/documents/${id}`)
}

export function deleteDocument(id: number): Promise<void> {
  return request.delete(`/documents/${id}`)
}

File: backend/app/schemas/document.py
"""文档相关校验/响应模型（Phase 3 知识库）。

输入与响应分离，避免泄露内部字段（AI 宪法第三章）。
"""
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class DocumentCreate(BaseModel):
    """上传资料时可选的元信息（如分类）。"""

    category: str | None = None


class DocumentPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    filename: str
    title: str
    file_type: str
    file_size: int
    chunk_count: int
    category: str
    status: str
    graph_status: str
    graph_error: str | None
    created_at: datetime


class DocumentChunkPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    document_id: int
    chunk_index: int
    content: str
    token_estimate: int


class DocumentDetail(DocumentPublic):
    chunks: list[DocumentChunkPublic] = []
